X10D/src/Core/IntrinsicUtility.cs

```diff
@@ -0,0 +1,222 @@
+using System.Runtime.Intrinsics.Arm;
+using System.Runtime.Intrinsics.X86;
+
+namespace X10D.Core;
+
+/// <summary>
+///     Provides utility methods for SIMD vector that is currently missing on common hardware instruction set.
+/// </summary>
+public static class IntrinsicUtility
+{
+    /// <summary>
+    ///     <br>Multiply packed 64-bit unsigned integer elements in a and b and truncate the results to 64-bit integer.</br>
+    ///     <br>Operation:</br>
+    ///     <code>
+    ///     dest[0] = lhs[0] * rhs[0];
+    ///     dest[1] = lhs[1] * rhs[1];
+    ///     </code>
+    /// </summary>
+    /// <param name="lhs">Left vector.</param>
+    /// <param name="rhs">Right vector.</param>
+    /// <returns></returns>
+    /// <remarks>API avaliable on SSE2, SSE3, SSSE3, SSE4.1, SSE4.2, AVX, AVX2, ARM NEON (untested) hardwares.</remarks>
+    [Pure]
+    [CLSCompliant(false)]
+    [MethodImpl(MethodImplOptions.AggressiveInlining | MethodImplOptions.AggressiveOptimization)]
+    public static Vector128<ulong> Multiply(Vector128<ulong> lhs, Vector128<ulong> rhs)
+    {
+        if (Sse2.IsSupported)
+        {
+            // https://stackoverflow.com/questions/17863411/sse-multiplication-of-2-64-bit-integers
+
+            Vector128<ulong> ac = Sse2.Multiply(lhs.AsUInt32(), rhs.AsUInt32());
+            Vector128<uint> b = Sse2.ShiftRightLogical(lhs, 32).AsUInt32();
+            Vector128<ulong> bc = Sse2.Multiply(b, rhs.AsUInt32());
+            Vector128<uint> d = Sse2.ShiftRightLogical(rhs, 32).AsUInt32();
+            Vector128<ulong> ad = Sse2.Multiply(lhs.AsUInt32(), d);
+            Vector128<ulong> high = Sse2.Add(bc, ad);
+            high = Sse2.ShiftLeftLogical(high, 32);
+
+            return Sse2.Add(high, ac);
+        }
+        else if (AdvSimd.IsSupported)
+        {
+            // https://stackoverflow.com/questions/60236627/facing-problem-in-implementing-multiplication-of-64-bit-variables-using-arm-neon
+
+            // Hasn't been tested since March 7th 2023 (Reason: Unavailable hardware)
+            var a = AdvSimd.ExtractNarrowingLower(lhs);
+            var b = AdvSimd.ExtractNarrowingLower(rhs);
+
+            var mul = AdvSimd.Multiply(rhs.AsUInt32(), AdvSimd.ReverseElement32(lhs).AsUInt32());
+
+            return AdvSimd.MultiplyWideningLowerAndAdd(AdvSimd.ShiftLeftLogical(mul.AsUInt64(), 32), a, b);
+        }
+
+        throw new PlatformNotSupportedException("Unsupported SIMD platform.");
+    }
+
+    /// <summary>
+    ///     <br>Multiply packed 64-bit unsigned integer elements in a and b and truncate the results to 64-bit integer.</br>
+    ///     <br>Operation:</br>
+    ///     <code>
+    ///     dest[0] = lhs[0] * rhs[0];
+    ///     dest[1] = lhs[1] * rhs[1];
+    ///     dest[2] = lhs[2] * rhs[2];
+    ///     dest[3] = lhs[3] * rhs[3];
+    ///     </code>
+    /// </summary>
+    /// <param name="lhs">Left vector.</param>
+    /// <param name="rhs">Right vector.</param>
+    /// <returns></returns>
```

Comment:
Same as above. The `returns` element on the xmldoc is empty. Also `PlatformNotSupportedException` is not documented